### DISCUSSION TITLE
LDAP Mapping in docker compose

### DISCUSSION THREAD

**Discussion by storagefire122** [semaphoreui/semaphore#1707] [Q&A] [answered]
Hi guys,
i am currently trying to install semaphore over docker with the docker-compose.yml. Unfortunatly i thought this is going to be easier. 
I made these settings in the docker-compose file, but the ldap mapping won't get configured inside config.json
Any ideas from your site?
That would be really helpful :-)

      SEMAPHORE_LDAP_MAPPINGS: '"dn": "distinguishedName","mail": "userPrincipalName","uid": "sAMAccountName","cn": "cn"'
BR

**Comment by bmichalkiewicz** [comment, marked as answer]
I have something like that: 
    environment:
      SEMAPHORE_LDAP_ACTIVATED: "yes"
      SEMAPHORE_LDAP_HOST: host
      SEMAPHORE_LDAP_PORT: "636"
      SEMAPHORE_LDAP_NEEDTLS: true
      SEMAPHORE_LDAP_DN_BIND: ""cn=user,cn=Users,dc=wtf,dc=df,dc=com,dc=pl""
      SEMAPHORE_LDAP_PASSWORD: "I_like_pancakes"
      SEMAPHORE_LDAP_DN_SEARCH: "ou=workers,dc=wtf,dc=df,dc=com,dc=pl"
      SEMAPHORE_LDAP_SEARCH_FILTER: "(&(sAMAccountName=%s)(memberOf=cn=ansible_admin,ou=AnsibleSemaphore,ou=APPS,ou=Groups,dc=wtf,dc=df,dc=com,dc=pl))"
      SEMAPHORE_LDAP_MAPPING_USERNAME: sAMAccountName

bounded with Microsoft AD

EDIT: works with 2.9.37

**Reply by storagefire122** [reply]
Great, thanks for your help - worked great :-)
